core/shared/src/main/scala-2/zio/internal/macros/LayerMacroUtils.scala

```diff
@@ -9,11 +9,22 @@ private[zio] trait LayerMacroUtils {
 
   type LayerExpr = c.Expr[ZLayer[_, _, _]]
 
+  private def verifyLayers(layers: Seq[c.Expr[ZLayer[_, _, _]]]): Unit =
+    for (layer <- layers) {
+      layer.tree match {
+        case apply @ Apply(_, Block(_, _) :: Nil) => {
+          c.abort(layer.tree.pos, "Cannot use by-name parameters")
```

Comment:
Can you tweak this error message to explain:

1. What's going wrong, in detail (a user might not have ever heard of "by-name" parameters)
2. Why it's going wrong, e.g., what problem is introduced by by-name parameters that prevents their invocation from appearing as input to provide
3. What the user can do to fix the problem

Reply:
I've created following error message:
```
──── ZLAYER ERROR ────────────────────────────────────────────────────

 Due to bug in Scala 2 compiler invoking methods with by-name parameters in provide/provideSome method does not work
 def createLayerByName(i: Int, x: => MyLayer): zio.ULayer[MyLayer]

 Bug can be workarounded in following ways:

 1. Assign method output to temporary variable
    ZLayer.provide(createLayerByName(...))
    ↓↓↓
    val temp = createLayerByName(...)
    ZLayer.provide(temp)

 2. Replace by-name parameter with lambda:
    def createLayerByName(i: Int, x: => MyLayer): zio.ULayer[MyLayer]
    ↓↓↓
    def createLayerByName(i: Int, x: () => MyLayer): zio.ULayer[MyLayer]

──────────────────────────────────────────────────────────────────────
```